README.md

```diff
@@ -31,6 +31,118 @@ Application-level tracing for Rust.
 structured, event-based diagnostic information. `tracing` is maintained by the
 Tokio project, but does _not_ require the `tokio` runtime to be used.
 
+## Usage
+
+### In libraries
+
+Libraries should only rely on the `tracing` crate and use the provided macros
+and types to collect whatever information might be useful to downstream consumers.
+
+```toml
+[dependencies]
+tracing = "0.1"
+```
+
+```rust
+use std::{error::Error, fmt};
+use tracing::{debug, error, info, span, warn, Level};
+
+#[tracing::instrument]
+pub fn shave(yak: usize) -> Result<(), Box<dyn Error + 'static>> {
+    debug!(excitement = "yay!", "hello! I'm gonna shave a yak.");
+    if yak == 3 {
+        warn!("could not locate yak!");
+        return Err(ShaveError::new(yak).into());
+    } else {
+        debug!("yak shaved successfully");
+    }
+    Ok(())
+}
+
+pub fn shave_all(yaks: usize) -> usize {
+    let span = span!(Level::TRACE, "shaving_yaks", yaks);
+    let _enter = span.enter();
+
+    info!("shaving yaks");
+
+    let mut yaks_shaved = 0;
+    for yak in 1..=yaks {
+        let res = shave(yak);
+        debug!(yak, shaved = res.is_ok());
+
+        if let Err(ref error) = res {
+            error!(yak, error = error.as_ref(), "failed to shave yak!");
+        } else {
+            yaks_shaved += 1;
+        }
+        debug!(yaks_shaved);
+    }
+
+    yaks_shaved
+}
+
+#[derive(Debug)]
+enum ShaveError {
+    MissingYak { yak: usize },
+}
+
+impl fmt::Display for ShaveError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ShaveError::MissingYak { yak } => write!(f, "shaving yak #{} failed!", yak),
+        }
+    }
+}
+
+impl Error for ShaveError {}
+
+impl ShaveError {
+    fn new(yak: usize) -> Self {
+        ShaveError::MissingYak { yak }
+    }
+}
```

Comment:
Up to you: we can replace this with https://github.com/dtolnay/thiserror or Snafu to reduce the emphasis on the error boilerplate.

Reply:
I'm a little iffy about showing the use of other crates in the basic/getting started examples — I don't want to create false impressions that those crates are required, or that tracing has specific integrations with them. 

What do you think about just getting rid of the custom error type in the example, and replacing it with something from `std`...perhaps `io::Error`?
```rust
io::Error::new(io::ErrorKind::Other, "shaving yak failed!")
```

Reply:
I think the custom error type with `io::ErrorKind::Other` is good.